### PR TITLE
Recognize Vector<T> constructors for small ints

### LINKED ISSUE #5116: Vector<T> constructor not recognized for ubyte, byte, short or ushort
The entry for .ctor (with a scalar argument) doesn't include ushort, but the other sub-int types are not recognized either, because the arg type is int, and therefore is not considered a match.
I am labeling this "Future" because I don't think these element types get significant use.

### PR DESCRIPTION
**Pull request by CarolEidt**
Fix #5116

### CODE REVIEW

**Comment by CarolEidt** [comment]
@dotnet-bot test Windows_NT x86 ryujit Checked Build and Test

**Comment by CarolEidt** [comment]
@sivarv @dotnet/jit-contrib PTAL.
We were not recognizing these scalar constructor of Vector<T> because the scalar argument always comes in as the evaluation type, not the unsigned or small integer type. These, as it happens, are the only SIMD intrinsics that take scalar operands of the element type.

File: src/jit/simd.cpp [hunk 1]
```diff
@@ -540,8 +540,8 @@ const SIMDIntrinsicInfo* Compiler::getSIMDIntrinsicInfo(CORINFO_CLASS_HANDLE*  i
                         assert(expectedArgType != TYP_UNDEF);
                         if (expectedArgType == TYP_UNKNOWN)
                         {
-                            // JIT maps uint/ulong type vars to TYP_INT/TYP_LONG.                            
-                            expectedArgType = (*baseType == TYP_UINT || *baseType == TYP_ULONG) ? genActualType(*baseType) : *baseType;
+                            // The type of the argument will be genActualType(*baseType).                            
+                            expectedArgType = genActualType(*baseType);
```
Comment (pgavlin) [inline src/jit/simd.cpp]: Just double-checking--by this point we've confirmed that `baseType` is a supported vector element type?

Comment (CarolEidt) [inline src/jit/simd.cpp]: Yes, baseType comes from the type handle (getBaseTypeAndSizeOfSIMDType).

Comment (CarolEidt) [inline src/jit/simd.cpp]: And TYP_UNKNOWN as an argument indicates an argument of type T (i.e. baseType).

**Comment by pgavlin** [comment]
LGTM, then.

**Comment by sivarv** [comment]
Looks good.

**Comment by AndyAyersMS** [comment]
@CarolEidt does this case get hit in the normal test pass (presumably yes since you found it with your new testing approach...)?

**Comment by CarolEidt** [comment]
Yes, this case was getting hit, but we didn't have any mechanism to verify that it was being recognized as an intrinsic. In fact, when/if I ever get my PR #5101 full tested and merged, I will then enable the validation for these as well.

**Pull request merged**